### ISSUE TITLE
Compile error on MinGW64 gcc 11.2

### ISSUE THREAD

**Issue by axojhf** [stephenberry/glaze#188]
```
[build] C:/include/glaze/thread/threadpool.hpp:40:39: error: invalid conversion from 'long long unsigned int' to 'HANDLE' {aka 'void*'} [-fpermissive]
[build]    40 |                SetThreadGroupAffinity(hndl, &affinity, nullptr);
[build]       |                                       ^~~~
[build]       |                                       |
[build]       |                                       long long unsigned int
```

https://github.com/stephenberry/glaze/blob/6d2a5961a05b9383f8d1c52885bebe94966eae99/include/glaze/thread/threadpool.hpp#L40

I can set this option in cmake
```cmake
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpermissive")
```
But I don't know if you need to be concerned about this

**Comment by mwalcott3** [comment]
Ohh this might actually be something to be concerned about. Thanks for bringing this up. MinGW has different threading models. I assume you are using posix threads. In that case, std::thread::native_handle is probably returning a pthread handle whereas  SetThreadGroupAffinity is expecting the windows HANDLE.

We could disable the processor group stuff  if __MINGW32__ is defined. We could probaly also check if the type returned by std::thread::native_handle is the same as HANDLE in an "if constexpr".

The group stuff is not strictly necessary anyway and was being used to utilize all available logical processors on windows systems with more than 64 logical processors that have multiple groups. Note that this is no longer needed in Windows 11 and Windows Server 2022 and up https://learn.microsoft.com/en-us/windows/win32/procthread/processor-groups

**Comment by mwalcott3** [comment]
Addressed in #189. Removed the usage of processor groups when using MinGW and made some other changes that seemed to be causing problems with MinGW. Added a MSYS2 MinGW gcc 12.2.0 workflow to make sure things work as expected on MinGW.

**Comment by axojhf** [comment]
Thanks
